projects/online/online/utils/gdb.py

```diff
@@ -121,3 +130,34 @@ def gracedb_factory(server: GdbServer, write_dir: Path) -> GraceDb:
     else:
         raise ValueError(f"Unknown GraceDB server: {server}")
     return GraceDb(service_url=server, write_dir=write_dir)
+
+
+def authenticate():
+    # TODO: don't hardcode keytab locations
+    subprocess.run(
+        [
+            "htgettoken",
+            "-v",
+            "-a",
+            "vault.ligo.org",
+            "-i",
+            "igwn",
+            "-r",
+            "aframe-1-scitoken",
+            "--scopes=gracedb.read",
+            "--credkey=aframe-1-scitoken/robot/ldas-pcdev12.ligo.caltech.edu",
+            "--nooidc",
+        ]
+    )
+
+    subprocess.run(
```

Comment:
I think the `kinit` and `htgettoken` processes should happen in the opposite order, in case the kerberos authentication has expired.

Reply:
good point. fixed